hail/src/main/scala/is/hail/expr/types/physical/PNDArray.scala

```diff
@@ -25,21 +25,21 @@ final case class PNDArray(elementType: PType, nDims: Int, override val required:
 
   override def codeOrdering(mb: EmitMethodBuilder, other: PType): CodeOrdering = throw new UnsupportedOperationException
 
-  val flags = new StaticallyKnownField(PInt32Required, (r, off) => Region.loadInt(representation.loadField(r, off, "flags")))
-  val offset = new StaticallyKnownField(
+  @transient lazy val flags = new StaticallyKnownField(PInt32Required, (r, off) => Region.loadInt(representation.loadField(r, off, "flags")))
```

Comment:
I'm not sure I understand what `StaticallyKnownField` is doing. It seems consistent with the other PType interfaces to split this into `val flagsPType = PInt32Required` and `def loadFlags(r: Code[Region], off: Code[Long])`, and similarly for the rest.

I also don't like that `ndarray.flags.load` takes the pointer to the ndarray, not to the flags field. To make this style of interface consistent, I think we'd need classes that package together a PType and a pointer. I do think that's the right interface to build, but we don't have it yet.

But this is well outside the scope of this PR. Approving.

Reply:
Yeah, this is only used on PNDArrays right now, and was really just to cut down on all the manual get fields, coercions, and loads I was doing everywhere. If there was a `loadFlags` method like you mention above, would you expect that one to take a pointer to the ndarray?

Reply:
Yes, I would expect it to have behave exactly as if `PNDArray` were really a subclass of `PBaseStruct`, and `loadFlags(r, off)` were a shorthand for `loadField(r, off "flags")`.